java/arcs/core/data/ParticleSpec.kt

```diff
@@ -22,5 +22,13 @@ data class ParticleSpec(
     val claims: List<Claim> = emptyList(),
     val checks: List<Check> = emptyList(),
     /** Indicates whether the particle is an isolated (non-egress) particle. */
-    val isolated: Boolean = false
-)
+    val isolated: Boolean = false,
+    /** Optional egress type of the particle. Must be null for isolated particles. */
```

Comment:
do we need an `isEgressType` boolean getter? what will happen if @egress annotation is present, but egress type isn't specified (a comment below suggests it is optional). there is such getter in the ts particle-spec class.

Reply:
I don't think so. You can specify `@egress` without an egress type. The `egressType` getter will return null in that case. If you want to know whether there's an egressType specified or not, use `egressType == null`